applications/crossbar/src/modules/cb_configs.erl

```diff
@@ -15,198 +12,164 @@
-%% request. Generally, this will involve stripping pvt fields and loading
-%% the resource into the resp_data, resp_headers, etc...
-%% @end
-%%--------------------------------------------------------------------
--spec get(cb_context:context(), path_token()) -> cb_context:context().
-get(Context, _) ->
-    Context.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verib is PUT, execute the actual action, usually a db save.
-%% @end
-%%--------------------------------------------------------------------
+validate(Context, ?HTTP_GET, ConfigId) ->
+    JObj = kapps_config_util:get_config(cb_context:account_id(Context), ConfigId),
+    crossbar_doc:handle_datamgr_success(set_id(ConfigId, JObj), Context);
+
+validate(Context, ?HTTP_DELETE, ConfigId) ->
+    Doc = case kapps_config_util:load_config_from_account(cb_context:account_id(Context), ConfigId) of
+              {ok, Document} -> Document;
+              _ -> set_id(ConfigId, kz_json:new())
+          end,
+    pass_validation(Context, Doc);
+
+validate(Context, ?HTTP_PUT, ConfigId) ->
+    validate(Context, ?HTTP_POST, ConfigId);
+
+validate(Context, ?HTTP_PATCH, ConfigId) ->
+    Parent = kapps_config_util:get_config(cb_context:account_id(Context), ConfigId),
+    save_diff(validate_diff(Context, ConfigId, Parent), ConfigId, Parent);
+
+validate(Context, ?HTTP_POST, ConfigId) ->
+    Parent = kapps_config_util:get_reseller_config(cb_context:account_id(Context), ConfigId),
+    validate_diff(Context, ConfigId, Parent).
+
 -spec put(cb_context:context(), path_token()) -> cb_context:context().
-put(Context, _) ->
-    crossbar_doc:save(Context).
-
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verib is POST, execute the actual action, usually a db save
-%% (after a merge perhaps).
-%% @end
-%%--------------------------------------------------------------------
+put(Context, ConfigId) -> post(Context, ConfigId).
+
 -spec post(cb_context:context(), path_token()) -> cb_context:context().
-post(Context, _) ->
-    crossbar_doc:save(Context).
-
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verib is PATCH, execute the actual action, usually a db save
-%% (after a merge).
-%% @end
-%%--------------------------------------------------------------------
--spec patch(cb_context:context(), path_token()) -> cb_context:context().
-patch(Context, _) ->
-    crossbar_doc:save(Context).
-
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%% If the HTTP verib is DELETE, execute the actual action, usually a db delete
-%% @end
-%%--------------------------------------------------------------------
+post(Context, ConfigId) ->
+    Parent = kapps_config_util:get_reseller_config(cb_context:account_id(Context), ConfigId),
+    save_diff(Context, ConfigId, Parent).
```

Comment:
during the validation complete config document is formed (e.g. with all parents merged), and this complete document is validated. but we need to save a difference to this document, therefore save_diff/3 instead of just save/2.